packages/core/src/lib/services/wallet-modules/wallet-modules.service.ts

```diff
@@ -189,6 +189,17 @@ export class WalletModules {
       });
     });
 
+    emitter.on("setActiveAccount", async ({ accounts, accountId }) => {
+      if (!accounts.length) {
+        return this.signOutWallet(module.id);
+      }
+
+      this.store.dispatch({
+        type: "SET_ACTIVE_ACCOUNT",
+        payload: { accountId, accounts },
+      });
+    });
```

Comment:
This emitter is really for events that occur within a wallet. We should move back to the `setActiveAccount` method on the `selector` object. We should only need the `accountId` as the store already contains the list of accounts.

I'm a little confused by the sign out logic we have here too. What's the reason for that?